benchmark/submit.py

```diff
@@ -0,0 +1,129 @@
+# Copyright ClusterHQ Inc.  See LICENSE file for details.
+"""
+Submit benchmark results to a server.
+"""
+
+from __future__ import print_function
+
+import json
+import sys
+
+from twisted.internet import endpoints
+from twisted.internet.defer import succeed
+from twisted.internet.task import react
+from twisted.python import usage
+from twisted.web import client, http
+from twisted.web.http_headers import Headers
+from twisted.web.iweb import IBodyProducer
+
+from zope.interface import implementer
+
+
+@implementer(IBodyProducer)
+class StringProducer(object):
+    def __init__(self, body):
+        self.body = body
+        self.length = len(body)
+
+    def startProducing(self, consumer):
+        consumer.write(self.body)
+        return succeed(None)
+
+    def pauseProducing(self):
+        pass
+
+    def stopProducing(self):
+        pass
+
+
+def submit(agent, result):
+    """
+    Post the given result using the given agent.
+
+    :param Agent agent: The HTTP client agent.
+    :param dict result: The JSON-compatible result.
+    :return: Deferred that fires when a reponse from the server is received.
+    """
+    body = StringProducer(json.dumps(result))
+    req = agent.request(
+        "POST",
+        "/v1/benchmark-results",
+        headers=Headers({'Content-Type': ['application/json']}),
+        bodyProducer=body
+    )
+
+    def get_response_body(response):
+        d = client.readBody(response)
+        d.addCallback(lambda body: (response, body))
+        return d
+
+    req.addCallback(get_response_body)
+
+    def process_response(response_and_body):
+        (response, body) = response_and_body
+        if response.code != http.CREATED:
+            print("Failed to submit the request", file=sys.stderr)
+            print("{} {}".format(response.code, response.phrase),
+                  file=sys.stderr)
+        else:
+            print("Successfully submitted the result")
+
+        json_body = json.loads(body)
+
+        if response.code != http.CREATED:
+            print("Message: {}".format(json_body['message']), file=sys.stderr)
+        else:
+            print("Assigned result ID: {}".format(json_body['id']))
+
+    req.addCallback(process_response)
+    return req
+
+
+class Options(usage.Options):
```

Comment:
Yeah, and in most cases the dcostrings seem to be very obvious and tautological like
```
Arguments for ``initialize-release`` script
```
I think that that does not add much of useful information and is prone to getting out-of-sync with reality (e.g. because of renaming or copying).